### ISSUE TITLE
Interpretation of average loss using --cb or --cb_explore

### ISSUE THREAD

**Issue by maxpagels** [VowpalWabbit/vowpal_wabbit#1667]
I ran the test described at https://github.com/VowpalWabbit/vowpal_wabbit/blob/b640cfde19f35a2bf33e67ba56f938473768475a/test/train-sets/ref/rcv1_raw_cb_explore.stderr using the same dataset and am able to reproduce the results. However, I am wondering how to interpret the average loss. If test the same dataset against a baseline model (`train-sets/rcv1_raw_cb_small.vw --cb_explore 2 --ngram 2 --skips 4 -b 24 -l 0.25 -t`), the average loss is 0.011200, compared to actually optimising a policy using the options in the test (`train-sets/rcv1_raw_cb_small.vw --cb_explore 2 --ngram 2 --skips 4 -b 24 -l 0.25`, giving a loss of 0.362700).

By all accounts, the trained policy should work better than a baseline policy that always chooses the same action, which is why I'm wondering information the average loss is actually conveying if it's higher compared to a baseline.

**Comment by maxpagels** [comment]
I'd be more than happy to contribute instructions to the wiki, if it's of use to others?

**Comment by JohnLangford** [comment]
The average loss is the average progressive validation loss (see here: http://hunch.net/~jl/projects/prediction_bounds/progressive_validation/coltfinal.pdf ) so it includes the loss on the first example when nothing has been learned.  

Updating the wiki would be great :-)

**Comment by maxpagels** [comment]
Note that using no exploration (`--cb` instead of `--cb-explore`) does the opposite: with `-t` the loss is much higher (as expected).

**Comment by JohnLangford** [comment]
I'm not following.  Why would -t make the classifier always predict label
1?  Is this an empirical observation?

-John

**Comment by maxpagels** [comment]
@JohnLangford thanks for the prompt response! I'm referring to `-t` as in test-only. I'm not passing in any model with `-i`, and all predications are of label 1:

```
§ vw rcv1_raw_cb_small.vw --cb_explore 2 --ngram 2 --skips 4 -b 24 -l 0.25 -t
Generating 2-grams for all namespaces.
Generating 4-skips for all namespaces.
only testing
Num weight bits = 24
learning rate = 0.25
initial_t = 0
power_t = 0.5
using no cache
Reading datafile = rcv1_raw_cb_small.vw
num sources = 1
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
0.050000 0.050000            1            1.0        1 1:0.975000      280
0.025000 0.000000            2            2.0        1 1:0.975000      598
0.012500 0.000000            4            4.0        1 1:0.975000      784
0.012500 0.012500            8            8.0        1 1:0.975000      850
0.012500 0.012500           16           16.0        1 1:0.975000      118
0.007813 0.003125           32           32.0        1 1:0.975000      166
0.008594 0.009375           64           64.0        2 1:0.975000      340
0.008594 0.008594          128          128.0        1 1:0.975000      610
0.010742 0.012891          256          256.0        2 1:0.975000      712
0.011133 0.011523          512          512.0        2 1:0.975000      424

finished run
number of examples = 1000
weighted example sum = 1000.000000
weighted label sum = 0.000000
average loss = 0.011200
total feature number = 440020
```

Nothing really surprising here, since I explicitly used `-t` without providing an initial classifier. But it's unclear to me how the average loss reported at the end is much higher if I actually do learning, e.g.:

```
§ vw rcv1_raw_cb_small.vw --cb_explore 2 --ngram 2 --skips 4 -b 24 -l 0.25
Generating 2-grams for all namespaces.
Generating 4-skips for all namespaces.
Num weight bits = 24
learning rate = 0.25
initial_t = 0
power_t = 0.5
using no cache
Reading datafile = rcv1_raw_cb_small.vw
num sources = 1
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
0.050000 0.050000            1            1.0        1 1:0.975000      280
0.025000 0.000000            2            2.0        1 2:0.975000      598
0.012500 0.000000            4            4.0        1 2:0.975000      784
0.250000 0.487500            8            8.0        1 2:0.975000      850
0.368750 0.487500           16           16.0        1 2:0.975000      118
0.245312 0.121875           32           32.0        1 1:0.975000      166
0.216406 0.187500           64           64.0        2 1:0.975000      340
0.216406 0.216406          128          128.0        1 2:0.975000      610
0.300195 0.383984          256          256.0        2 2:0.975000      712
0.333984 0.367773          512          512.0        2 1:0.975000      424

finished run
number of examples = 1000
weighted example sum = 1000.000000
weighted label sum = 0.000000
average loss = 0.362700
total feature number = 440020
```

This seems counterintuitive to me; I'd expect a classifier that isn't trained on data at always selects the same label to report a loss higher than a classifier that's been trained on data. I'm probably getting something wrong, but since the situation might be confusing to others, I'd be happy to revise the documentation if I knew exactly what was going on.

**Comment by JohnLangford** [comment]
Closing — the question about interpreting average loss with contextual bandits was discussed in the comments. The key point is that progressive validation loss with IPS can appear counterintuitive because it reweights by inverse propensity, which can inflate loss estimates for low-probability actions. The CB documentation on the wiki has been expanded since this was filed.
